testovac/submit/views.py

```diff
@@ -86,14 +90,40 @@ def view_submit(request, submit_id):
     submit = get_object_or_404(Submit.objects.select_related('receiver'), pk=submit_id)
     user_has_admin_privileges = submit.receiver.has_admin_privileges(request.user)
 
-    if submit.user != request.user and not user_has_admin_privileges:
+    def get_submit_picture(is_positive, submit_id):
+        if is_positive:
+            dirn = 'positive/'
+        else:
+            dirn = 'negative/'
+        files = map(os.path.basename, glob(os.path.join(settings.STATIC_ROOT,'gifs',dirn,'*')))
+        if files == []:
+            return None
+        files.sort()
+        numbers = list(filter(lambda n: n[1]<submit_id, enumerate(map(lambda n: int(n.split('.')[0]), files))))
+        if numbers == []:
+            return None
+        picture_id = submit_id % len(numbers)
+        return reduce(urlparse.urljoin, ['/static/','gifs/', dirn,files[numbers[picture_id][0]]])
```

Comment:
Toto mi pride ako celkom zverina.
Chcelo by to nejaky komentar ze co to ma presne robit, aby sa to vobec dalo zreviewovat.
A potom to bude treba upratat.
Snazit sa manualne citat static nie je dobry napad a uz toboz nie pomocou glob. Minimalne pre to ze statics moze servovat uplne iny server ako appku.
Tiez si nie som uplne isty tym pouzitim urljoin.